### ISSUE TITLE
Looking for a way to start a DOS command with parameters directly

### ISSUE THREAD

**Issue by tonypdmtr** [joncampbell123/dosbox-x#2758]
I have old DOS utilities that do some file conversions.  They don't run under Win7-64, and there is no source available.

Is there a way to run them under `dosbox-x` but without starting the `dosbox-x` console, and then typing the command manually?

I want this whole thing to be put in a Windows batch file, e.g., do something, run the DOS-era command using `dosbox-x`, and continue with the rest of the batch file outside of `dosbox-x`.  So, `dosbox-x` would be needed just to run my old DOS-era program specified on the command line that starts `dosbox-x`, and, upon completion of that program, exit back to Windows console. Like so:

```
batch_command1
dosbox-x my_dos_app.exe my_dos_app_parm1 my_dos_app_parm2 ...
batch_command3
```

Possible?

**Comment by Wengier** [comment]
Try this:

```
dosbox-x -silent -c “myapp.exe …”
```

**Comment by tonypdmtr** [comment]
Tried it with some modifications (dot for current directory, and 2nd command `exit`), and it worked:
`dosbox-x . -c "myapp.exe ..." -c "exit"`

Any way to capture the output of the utility normally shown on the screen?  Appending `> output` did not do it.

**Comment by Wengier** [comment]
Try this setting:

```
logfile = output.log
log console = quiet
```
Then look at output.log.

P.S. The -silent option will hide the DOSBox-X window completely and may be useful for fully automated task(s). But if your program does need some interactions then do not use this option.

**Comment by tonypdmtr** [comment]
Thank you.

Anyway to specify the logfile on the command line?  If I have to do this batch for different files, it's a pain to have to edit the configuration file before each run.

It would be nice, however, to have a single command-line switch specifically for running console-only apps.  This would run the application (and exit immediately) using as default the `dosbox-x` stdin (the Windows console or stdin if specified) and stdout (the Windows console or stdout if specified).  Then running any DOS command-line only app would as simple as:

`dosbox-x -console dos_command parm1 parm2 ... < optional_stdin > optional_stdout`

If stdin and/or stdout are not redirected, then use the actual Windows console.

**Comment by Wengier** [comment]
Yes, use -set option for this. Check out the Wiki for more information about command-line options.

**Comment by tonypdmtr** [comment]
Couldn't make the log file work as I need it.

It shows only dosbox-x related log but not my program's output.  It ends with:
```
Screen report: Method 'Win98base' (-1.000 x -1.000 pixels) at (0.000 x 0.000) (-1.000 x -1.000 mm) (-0.039 x -0.039 in) (-1.000 x -1.000 DPI)
--END OF LOG--
```

If I remove the `-silent` option I can see my program's output inside the dosbox-x window, so it's there for sure.  It simply isn't copied out to the log file.

**Comment by Wengier** [comment]
```
dosbox-x -set logfile = output.log -set “dos log console = quiet”
```

**Comment by tonypdmtr** [comment]
That worked, thank you.

(Still, maybe the developers could consider a simple, one-option alternative, like `-run` that will directly work with the Windows console or given stdin/stdout and make the DOS app run as if it were native to whatever system dosbox-x is on.  I looked at EMU2 and it's exactly what I mean but, unfortunately it seems to only run under Linux/Cygwin, not native Windows.)
